cmake/Modules/FindNCCL.cmake

```diff
@@ -57,15 +57,15 @@ if(NCCL_FOUND)  # obtaining NCCL version and some sanity checks
   include(CheckCXXSymbolExists)
   check_cxx_symbol_exists(NCCL_VERSION_CODE nccl.h NCCL_VERSION_DEFINED)
 
-  if (NCCL_VERSION_DEFINED)
+  # this condition check only works for non static NCCL linking
```

Comment:
@atalman It actually does work for static NNCL too. You just need to link against cudart explicitely, i.e.  add `CUDA::cudart` to the `LINK_LIBRARIES` (Line 77)